### ISSUE TITLE
code confusing me in def._sample_negative

### ISSUE THREAD

**Issue by ck6698000** [tfzhou/ContrastiveSeg#60]
as the code proposed in loss_contrast_mem.py

```
def _sample_negative(self, Q): 
        class_num, cache_size, feat_size = Q.shape
        X_ = torch.zeros((class_num * cache_size, feat_size)).float().cuda()
        y_ = torch.zeros((class_num * cache_size, 1)).float().cuda()
        sample_ptr = 0
        for ii in range(class_num):
            if ii == 0: continue   ####?
            this_q = Q[ii, :cache_size, :]
            X_[sample_ptr:sample_ptr + cache_size, ...] = this_q
            y_[sample_ptr:sample_ptr + cache_size, ...] = ii
            sample_ptr += cache_size

        return X_, y_

```
I'm wondering  what `if ii == 0: continue` is  for? The function of this line seems to be jump over the class 0 in memory bank, and adding much zeros in the final loop. A possible explaination shall be that class0 is bg, but I check the code and I suppose not to be this case. So can anyone explain this for me? Thanks!

**Comment by tfzhou** [comment]
For most semantic segmentation datasets Cityscapes, ADE, PASCAL-Context, class 0 denotes the non-labeled pixels. We ignore them in contrastive loss computation.

**Comment by nysp78** [comment]
@tfzhou  So, the class 0 is the ignore index and not the background class? I mean, for the background class is applied also the contrastive loss. Right?

**Comment by ck6698000** [comment]
> For most semantic segmentation datasets Cityscapes, ADE, PASCAL-Context, class 0 denotes the non-labeled pixels. We ignore them in contrastive loss computation.

thanks for reply!

**Comment by zhangyuxuan321** [comment]
> Reference in

Have you solved this problem yet? The number 0 also means the background class in my segmentation task. I can't figure out that whether to use it in author's contrastive method.
